File: 387-first-unique-character-in-a-string/387-first-unique-character-in-a-string.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        int n=s.length();
        //Store Frequency of all characters
        vector <int> fre(26,0);
        for(int i=0;s[i]!='\0';++i)
            fre[s[i]-'a']+=1;
        //Find Non-Repeating Character
        for(int i=0;s[i]!='\0';++i)
            if(fre[s[i]-'a']==1)
                return i;
        
        return -1;
    }
    
};

File: 387-first-unique-character-in-a-string/387-first-unique-character-in-a-string.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        frequency = {}
        for i in s:
            if i not in frequency:
                frequency[i] = 1
            else:
                frequency[i] +=1
        for i in range(len(s)):
            if frequency[s[i]] == 1:
                return i
        return -1